src/main/windows/windowManager.ts

```diff
@@ -146,35 +157,44 @@ export class WindowManager {
         if (this.callsWidgetWindow) {
             this.switchServer(this.callsWidgetWindow.getServerName());
             this.mainWindow?.focus();
-            const currentView = this.viewManager?.getCurrentView();
-            currentView?.view.webContents.send(DESKTOP_SOURCES_MODAL_REQUEST);
+            this.callsWidgetWindow.getMainView().view.webContents.send(DESKTOP_SOURCES_MODAL_REQUEST);
         }
     }
 
     handleCallsWidgetChannelLinkClick = () => {
         log.debug('WindowManager.handleCallsWidgetChannelLinkClick');
+
         if (this.callsWidgetWindow) {
             this.switchServer(this.callsWidgetWindow.getServerName());
             this.mainWindow?.focus();
-            const currentView = this.viewManager?.getCurrentView();
-            currentView?.view.webContents.send(BROWSER_HISTORY_PUSH, this.callsWidgetWindow.getChannelURL());
+            this.callsWidgetWindow.getMainView().view.webContents.send(BROWSER_HISTORY_PUSH, this.callsWidgetWindow.getChannelURL());
         }
     }
 
-    handleCallsError = (event: IpcMainEvent, msg: CallsErrorMessage) => {
+    handleCallsError = (_: string, msg: CallsErrorMessage) => {
         log.debug('WindowManager.handleCallsError', msg);
+
         if (this.callsWidgetWindow) {
             this.switchServer(this.callsWidgetWindow.getServerName());
             this.mainWindow?.focus();
             this.callsWidgetWindow.getMainView().view.webContents.send(CALLS_ERROR, msg);
         }
     }
 
-    handleCallsLinkClick = (_: IpcMainEvent, msg: CallsLinkClickMessage) => {
+    handleCallsLinkClick = (_: string, msg: CallsLinkClickMessage) => {
         log.debug('WindowManager.handleCallsLinkClick with linkURL', msg.link);
-        this.mainWindow?.focus();
-        const currentView = this.viewManager?.getCurrentView();
-        currentView?.view.webContents.send(BROWSER_HISTORY_PUSH, msg.link);
+
+        if (this.callsWidgetWindow) {
```

Comment:
This seems like double check, is that intentional?

Reply:
@cpoile What do you mean by double check? It's the usual "if widget window exists do something" pattern. We could use optional chaining but I feel this is clearer. 

Unless you mean if we get there it's likely there's a widget open but I am not sure we can guarantee it won't be closed once we reach that as there's probably some asynchronous logic involved.

Reply:
Yah, the second. Sure, that makes sense, and it doesn't hurt. 👍 
Although... if all of the handlers do this check, we can move it into the `genCallsEventHandler` HOF.

Reply:
I've thought about that but it would break older plugin versions where the global widget is not available so I left it where strictly necessary to keep backwards compatibility. For example, all the logic to get screen sharing sources should work regardless of the global widget. At some point hopefully we can deprecate all that and simplify.

Reply:
Ah, ok. Cool, thanks!